### PR TITLE
Big update

### PR DESCRIPTION
**Pull request by kddeisz**
This PR does a couple of things. It:

1. Drops support for ruby 2.0.x. It's been dropped by ruby for a very long time, and there's no reason to support it anymore. If people need it, they can use an older version of humidifier.
2. Checks the docs into the `docs` directory. This greatly simplifies our docs workflow and will hopefully ensure that they always get updated. 
3. Drops the `Utils::enumerable_to_h` method which was needed to backport the `to_h` method to ruby 2.0.

### CODE REVIEW

File: lib/humidifier/version.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 module Humidifier
   # Gem version
-  VERSION = '1.0.6'.freeze
+  VERSION = '1.2.1'.freeze
```
Comment (meagonqz) [inline lib/humidifier/version.rb]: 👀 ? why patch .1

**Comment by meagonqz** [comment]
What's the process now for updating the docs? Are they still hosted by github on `gh-pages`?

File: lib/humidifier/version.rb (hunk 1 above)
Comment (kddeisz) [inline lib/humidifier/version.rb]: To stay in line with the AWS CloudFormation resource specification version (so people know which version is which)

**Comment by kddeisz** [comment]
They'll still be hosted by github pages, but it's just going to be off the master branch `docs` directory. (This wasn't an option when humidifier was built originally.)

File: README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Humidifier allows you to build AWS CloudFormation (CFN) templates programmatical
 
 For the full docs, go to [https://localytics.github.io/humidifier/](http://localytics.github.io/humidifier/). For local development instructions, see the [Development](https://localytics.github.io/humidifier/#label-Development) section.
```
Comment (meagonqz) [inline README.md]: It looks like the Development instructions need to be updated to specify Ruby 2.1 as well if they aren't already (big PR)

https://localytics.github.io/humidifier/#label-Development

**Pull request merged**